qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -202,6 +216,11 @@ def save_account(
                 'Invalid IBM Quantum Experience token '
                 'found: "{}" of type {}.'.format(token, type(token)))
 
+        if hgp:
+            # Set the hub/group/project info for the `credentials`.
+            hgp_dict = get_provider_as_dict(hgp)
+            kwargs.update(hgp_dict)
```

Comment:
Setting h/g/p in `Credential` has a lot of implication in terms of saving it, due to how things were done before. It's probably easier to just pass in a new `default_provider` parameter.